tools/benchmark/src/mocha/memoryTestRunner.ts

```diff
@@ -337,54 +343,81 @@ export function benchmarkMemory(testObject: IMemoryTestObject): Test {
 				samples: [],
 				variance: Number.NaN,
 			};
+
 			do {
 				await testObject.beforeIteration?.();
-
 				global.gc();
-				benchmarkStats.samples.before.memoryUsage.push(process.memoryUsage());
-				benchmarkStats.samples.before.heap.push(v8.getHeapStatistics());
-				benchmarkStats.samples.before.heapSpace.push(v8.getHeapSpaceStatistics());
+				sample.before.memoryUsage.push(process.memoryUsage());
+				sample.before.heap.push(v8.getHeapStatistics());
+				sample.before.heapSpace.push(v8.getHeapSpaceStatistics());
 
 				global.gc();
 				await testObject.run();
 
 				await testObject.afterIteration?.();
 
 				global.gc();
-				benchmarkStats.samples.after.memoryUsage.push(process.memoryUsage());
-				benchmarkStats.samples.after.heap.push(v8.getHeapStatistics());
-				benchmarkStats.samples.after.heapSpace.push(v8.getHeapSpaceStatistics());
 
-				benchmarkStats.runs++;
+				sample.after.memoryUsage.push(process.memoryUsage());
+				sample.after.heap.push(v8.getHeapStatistics());
+
+				sample.after.heapSpace.push(v8.getHeapSpaceStatistics());
+
+				runs++;
 
 				const heapUsedArray: number[] = [];
-				for (let i = 0; i < benchmarkStats.samples.before.memoryUsage.length; i++) {
+				for (let i = 0; i < sample.before.memoryUsage.length; i++) {
 					heapUsedArray.push(
-						benchmarkStats.samples.after.memoryUsage[i].heapUsed -
-							benchmarkStats.samples.before.memoryUsage[i].heapUsed,
+						sample.after.memoryUsage[i].heapUsed -
+							sample.before.memoryUsage[i].heapUsed,
 					);
 				}
 				heapUsedStats = getArrayStatistics(heapUsedArray, options.samplePercentageToUse);
 
 				// Break if max elapsed time passed, only if we've reached the min sample count
 				if (
-					benchmarkStats.runs >= options.minSampleCount &&
+					runs >= options.minSampleCount &&
 					timer.toSeconds(startTime, timer.now()) > options.maxBenchmarkDurationSeconds
 				) {
 					break;
 				}
 			} while (
-				benchmarkStats.runs < options.minSampleCount ||
+				runs < options.minSampleCount ||
 				heapUsedStats.marginOfErrorPercent > options.maxRelativeMarginOfError
 			);
-			benchmarkStats.stats = heapUsedStats;
+
+			benchmarkStats.customDataFormatters.Iteration = (value): string =>
```

Comment:
Let's move this one to be next to where we set the value, and fix it to read "Iterations", right now it would not be applied because of the key mismatch.